projects/js-packages/publicize-components/src/components/media-section/index.js

```diff
@@ -40,34 +44,53 @@ const ADD_MEDIA_LABEL = __( 'Set Social Image', 'jetpack' );
 const REPLACE_MEDIA_LABEL = __( 'Replace Social Image', 'jetpack' );
 const REMOVE_MEDIA_LABEL = __( 'Remove Social Image', 'jetpack' );
 
-const ALLOWED_MEDIA_TYPES = [ 'image/jpeg', 'image/png' ];
-
 /**
  * Wrapper that handles media-related functionality.
  *
  * @returns {object} The media section.
  */
 export default function MediaSection() {
+	const [ showNotice, setShotNotice ] = useState( false );
 	const { attachedMedia, updateAttachedMedia } = useAttachedMedia();
+	const { connections } = useSocialMediaConnections();
+	const enabledConnections = connections.filter( connection => connection.enabled );
+
+	const mediaValidator = useMemo( () => new MediaValidator( enabledConnections ), [
+		enabledConnections,
+	] );
 
 	const mediaObject = useSelect( select =>
 		select( 'core' ).getMedia( attachedMedia[ 0 ] || null, { context: 'view' } )
 	);
-
 	const { mediaWidth, mediaHeight, mediaSourceUrl } = getMediaDetails( mediaObject );
 
+	useEffect( () => {
+		// Removes selected media if connection change results in invalid image
+		if (
+			mediaObject &&
+			! mediaValidator.isImageValid( mediaObject.media_details?.filesize, mediaObject.mime_type )
+		) {
+			updateAttachedMedia( [] );
+			setShotNotice( true );
+		}
+	}, [ updateAttachedMedia, mediaObject, mediaValidator ] );
```

Comment:
This is an area I'm not sure about. I like the idea of a custom hook, but I'm not sure how far to go with it. We could move the use of `useSocialMediaConnections` into it, but should we pass in `mediaObject` and `updateAttachedMedia` or use those hooks within it?

Perhaps we could also have a `useMediaObject` hook or add that to `useAttachedMedia`, but I think we should probably keep custom hooks as simple as possible, in a similar way to classes having a single responsibility.

Either way you could memoize the result of validating the object with the dependencies of `[ updateAttachedMedia, mediaObject, enabledConnections ]` and then it should only be re-run when the media changes, or the enabled connections.

One thing I've noticed is that I think [the `enabledConnections` (here and in the hook) won't be stable](
https://github.com/Automattic/jetpack/tree/e88dcaf15cf2004fe1813566b05fa733f25ce2fd/projects/js-packages/publicize-components/src/hooks/use-social-media-connections/index.js#L17), so there's chance that could invalidate things on every render. It's not the end of the world I don't think, but it could lead to some unnecessary computation.